projects/igniteui-angular/src/lib/accordion/accordion.component.ts

```diff
@@ -0,0 +1,339 @@
+import { AfterContentInit, AfterViewInit, Component, ContentChildren, EventEmitter,
+    HostBinding, Input, OnDestroy, Output, QueryList } from '@angular/core';
+import { fromEvent, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { ACCORDION_NAVIGATION_KEYS } from '../core/utils';
+import { IExpansionPanelCancelableEventArgs,
+    IExpansionPanelEventArgs, IgxExpansionPanelBase } from '../expansion-panel/expansion-panel.common';
+import { IgxExpansionPanelComponent } from '../expansion-panel/expansion-panel.component';
+import { ToggleAnimationSettings } from '../expansion-panel/toggle-animation-component';
+
+interface IAccordionEventArgs extends IExpansionPanelEventArgs {
+    owner: IgxAccordionComponent;
+    panel: IgxExpansionPanelBase;
+}
+
+interface IAccordionCancelableEventArgs extends IExpansionPanelCancelableEventArgs {
+    owner: IgxAccordionComponent;
+    panel: IgxExpansionPanelBase;
```

Comment:
Same as the above but the comment here should say that it's the panel which is currently expanding or collapsing.